cinema/permissions.py

```diff
@@ -0,0 +1,11 @@
+from rest_framework.permissions import BasePermission, SAFE_METHODS
+
+
+class IsAdminOrIfAuthenticatedReadOnly(BasePermission):
+    def has_permission(self, request, view):
+        is_authenticated = request.user and request.user.is_authenticated
+        if request.method in SAFE_METHODS:
+            return is_authenticated
+        if request.method == "POST" and view.basename == "order":
+            return is_authenticated
+        return request.user and request.user.is_staff
```

Comment:
The logic for allowing authenticated users to create orders seems correct, but ensure it aligns with the task requirements. According to the task, only authenticated users should be able to create orders, and deletion should be prohibited even for administrators outside the admin panel. Double-check that this logic is correctly implemented in your views and permissions.